Tasks/SonarQubePostTest/CodeAnalysisFilePathComputation.ps1

```diff
@@ -0,0 +1,275 @@
+
+        LoadClientOMAssemblies
+        LoadUploaderAssembly
+      
+        $errorMessage = ""             
+        $postProcessResult = [Microsoft.VisualStudio.Services.CodeAnalysis.PostProcessor.PostProcessor]::ProcessCodeAnalysisResults(`
+            $vssConnection, $teamProject, $repoId, $sourceBranch, $sonarReportProcessedFilePath, [ref]$errorMessage)
+        
+        if(!$postProcessResult)
+        {            
+            throw "failed to process the report file $sonarReportProcessedFilePath. $errorMessage";
+        }
+        else
+        {
+            Write-Verbose "Processing of $sonarReportProcessedFilePath file is completed"
+        }
+    }
+    else
+    {
+        throw "Could not find file $sonarReportProcessedFilePath";
+    }   
+}
+
+
+
+function LoadClientOMAssemblies
+{                    
+        $tfsClientOMDir = $env:AGENT_SERVEROMDIRECTORY        
+
+        Write-Verbose "Loading TFS client object model assemblies packaged with the build agent"
+                
+        [System.Reflection.Assembly]::LoadFile([System.IO.Path]::Combine($tfsClientOMDir,"Microsoft.TeamFoundation.SourceControl.WebApi.dll")) | Out-Null
+        [System.Reflection.Assembly]::LoadFile([System.IO.Path]::Combine($tfsClientOMDir,"Microsoft.TeamFoundation.Common.dll")) | Out-Null            
+        [System.Reflection.Assembly]::LoadFile([System.IO.Path]::Combine($tfsClientOMDir,"Microsoft.TeamFoundation.Core.WebApi.dll")) | Out-Null            
+        [System.Reflection.Assembly]::LoadFile([System.IO.Path]::Combine($tfsClientOMDir,"Microsoft.VisualStudio.Services.Client.dll")) | Out-Null            
+        [System.Reflection.Assembly]::LoadFile([System.IO.Path]::Combine($tfsClientOMDir,"Microsoft.VisualStudio.Services.CodeReview.Common.dll")) | Out-Null            
+        [System.Reflection.Assembly]::LoadFile([System.IO.Path]::Combine($tfsClientOMDir,"Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.dll")) | Out-Null            
+        [System.Reflection.Assembly]::LoadFile([System.IO.Path]::Combine($tfsClientOMDir,"Microsoft.TeamFoundation.SourceControl.WebApi.dll")) | Out-Null            
+        [System.Reflection.Assembly]::LoadFile([System.IO.Path]::Combine($tfsClientOMDir,"Microsoft.VisualStudio.Services.Common.dll")) | Out-Null            
+        [System.Reflection.Assembly]::LoadFile([System.IO.Path]::Combine($tfsClientOMDir,"Microsoft.VisualStudio.Services.WebApi.dll")) | Out-Null            
+        [System.Reflection.Assembly]::LoadFile([System.IO.Path]::Combine($tfsClientOMDir,"Newtonsoft.Json.dll"))                                  
+}
+
+function LoadUploaderAssembly
+{        
+        $currentDir = Get-Location        
+        [System.Reflection.Assembly]::LoadFile("$currentDir\Microsoft.VisualStudio.Services.CodeAnalysis.PostProcessor.dll") | Out-Null           
+}
+
+function ProcessSonarCodeAnalysisReport
+{    
+    $sonarReportFilePath = GetSonarReportFilePath
```

Comment:
I changed the path computation so that all the methods use common helpers instead of calculating paths on their own